### PR TITLE
43 parse location block in config file

### PR DESCRIPTION
**Pull request by forbidden-arts**
i commented out the old main, just so we don't lose it and that i don't have to provide the code in here.

this will handle multiple location blocks over multiple servers.

earlier, @rosamakinen and i had a strange error (segfault) but we cannot reproduce it, even by reverting. not sure what to make of it, but -fsanitize doesn't show anything, so i (we) are flummoxxed.

### CODE REVIEW

**Comment by merituulie** [comment]
If there is a segfault, I wouldnt push this to main. Did you try to create the same segfault with the original code in main to see if its something already in the main or in this branch?

Also, no commented code into the main, we can get old shit back from version control. Comments and commented code deprecates and makes the code less readable.

**Comment by forbidden-arts** [comment]
> If there is a segfault, I wouldnt push this to main. Did you try to create the same segfault with the original code in main to see if its something already in the main or in this branch?

tried and failed. 
> 
> Also, no commented code into the main, we can get old shit back from version control. Comments and commented code deprecates and makes the code less readable.
> 
> Also Im missing the idea of stopping the socket / running server in the main to just print out server configurations? I wouldnt make the main to not run existing functionality for the rest of the team to be able to test their stuff :)



> If there is a segfault, I wouldnt push this to main. Did you try to create the same segfault with the original code in main to see if its something already in the main or in this branch?
> 
> Also, no commented code into the main, we can get old shit back from version control. Comments and commented code deprecates and makes the code less readable.
> 
> Also Im missing the idea of stopping the socket / running server in the main to just print out server configurations? I wouldnt make the main to not run existing functionality for the rest of the team to be able to test their stuff :)

1. tried and failed. even loading up the hash from where we started this morning does not reproduce the error. and i haven't pulled on my laptop, which also no longer has the error. seriously weird.
2 & 3. done

File: src/main.cpp [hunk 1]
```diff
@@ -125,12 +125,46 @@ void runServer(Server *server)
...
+	}
+}
+
 int main(int argc, char **argv)
 {
 	if (argc != 2 || !argv[1])
 	{
-		std::cerr << "The program needs one configuration file as input\n";
-		return 1;
+		dansTestFunc();
```
Comment (merituulie) [inline src/main.cpp:166]: I mean, sure, if you promise to put the original valid error message back once you're done with testing... :D

Comment (merituulie) [reply src/main.cpp:166]: Also the return as 1 should be changed back when you're done, as returning as 0 would indicate successful program run!

**Review by merituulie — APPROVED** [review]

**Comment by merituulie** [comment]
https://www.google.com/url?sa=i&url=https%3A%2F%2Fgiphy.com%2Fgifs%2Fso-you-think-you-can-dance-excited-jason-derulo-lets-go-3oEduYKjPVbx6vHiJW&psig=AOvVaw13B-SNjsYwT042iiaeJUYd&ust=1705578023432000&source=images&cd=vfe&ved=0CBIQjRxqFwoTCPib5rKr5IMDFQAAAAAdAAAAABAE

**Review by rosamakinen — APPROVED** [review]
okkurrrrr

**Pull request merged**